src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -521,8 +521,7 @@ private static boolean shouldUpdateUninitializedVariables(DetailAST ast) {
      * @return true if token is LITERAL_IF and there is an {@code else} following, else false
      */
     private static boolean isIfTokenWithAnElseFollowing(DetailAST ast) {
-        return ast.getType() == TokenTypes.LITERAL_IF
-                && ast.getLastChild().getType() == TokenTypes.LITERAL_ELSE;
+        return ast.getLastChild().getType() == TokenTypes.LITERAL_ELSE;
```

Comment:
Method name should be changed

Reply:
Done, changed to `isWithElseFollowing`

Reply:
Let's just remove method completely, we only have single condition now.

Reply:
Done.